app/models/repository.rb

```diff
@@ -169,7 +201,7 @@ def self.create_or_update!(repo)
     end
 
     # Finally remove the tags that are left and return the repo.
-    repository.tags.where(name: to_be_deleted_tags).find_each(&:delete_and_update!)
+    repository.tags.where(name: to_be_deleted_tags).find_each { |t| t.delete_and_update!(portus) }
```

Comment:
This is because there are two paths when it comes to deleting a repo/tag:

1. A delete event has come.
2. The user actually clicks on the browser.

On the second case, we remove images on the registry, and if we are successful we remove them from the DB. The problem with that is that by "remove images on the registry" I mean that there will be a delete notification for this same registry :smile:. This is why I've added a `marked` attribute: on the second case we "mark" tags that are to be deleted, as a way to lock the delete event from removing the tags that we want to remove. We do this because in this case we know who performed the action. In the case of the delete notification, no `actor` is given, so we don't know who effectively removed the tag.

This line of code you're pointing at refers to the first case: a delete notification. Since we don't know who performed the action, we set "portus" there. It's a similar situation as when crono creates repositories in the DB when Portus was not properly sync'ed by the web notification.